### PR TITLE
Migrating release scripts to the Makefile

### PR DESCRIPTION
**Pull request by xiam**
This PR intends to close https://github.com/getlantern/lantern/issues/2362.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,6 +23,14 @@ PACKAGE_VENDOR := Brave New Software Project, Inc
 PACKAGE_MAINTAINER := Lantern Team <team@getlantern.org>
 PACKAGE_URL := https://www.getlantern.org
 
+GH_USER := getlantern
+#GH_USER := xiam
```
Comment (xiam) [inline Makefile:27]: I still need this here for testing on another namespace without worrying of messing our official repo.

File: Makefile [hunk 2]
```diff
@@ -18,6 +23,14 @@ PACKAGE_VENDOR := Brave New Software Project, Inc
...
+
+GH_RELEASE_REPOSITORY := flashlight-build
+
+S3_BUCKET := lantern
+#S3_BUCKET := xiam-lantern-test-1
```
Comment (xiam) [inline Makefile:32]: This was also for testing.

File: Makefile [hunk 3]
```diff
@@ -253,10 +287,59 @@ binaries: docker genassets linux windows darwin
...
+		SUFFIX=$$(echo "$$NAME" | sed s/$$BASE_NAME//g) && \
+		VERSIONED=lantern-installer-$$TAG$$SUFFIX && \
+		echo "Copying $$VERSIONED" && \
+		$(S3CMD) -q cp s3://$(S3_BUCKET)/$$NAME s3://$(S3_BUCKET)/$$VERSIONED; \
+	done && \
+	echo "Uploading Windows binary for auto-updates" && \
+	$(RUBY) ./installer-resources/tools/uploadghasset.rb $(GH_USER) $(GH_RELEASE_REPOSITORY) $$TAG update_windows_386.bz2 && \
+	echo "Uploading OSX binary for auto-updates" && \
+	$(RUBY) ./installer-resources/tools/uploadghasset.rb $(GH_USER) $(GH_RELEASE_REPOSITORY) $$TAG update_darwin_amd64.bz2 && \
+	echo "Uploading Linux binaries for auto-updates" && \
```
Comment (xiam) [inline Makefile:315]: Linux QA and upload assets were missing.

File: copyexecutables.bash [hunk 4]
```diff
@@ -1,38 +0,0 @@
-#!/usr/bin/env bash
```
Comment (xiam) [inline copyexecutables.bash:1]: File was moved to `installer-resources/tools/`.

File: createrelease.rb [hunk 5]
```diff
@@ -1,37 +0,0 @@
-#!/usr/bin/env ruby
```
Comment (xiam) [inline createrelease.rb:1]: File was moved to `installer-resources/tools/`.

File: installer-resources/tools/createrelease.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  "target_commitish" => "",
+  "name"             => release_name,
+  "body"             => "",
+  "draft"            => false,
+  "prerelease"       => false,
+}.to_json
+
+response = http.request(request)
+
+if response.body.include? "already_exists"
```
Comment (xiam) [inline installer-resources/tools/createrelease.rb:31]: If the user attempts to create a release that already exists, he probably wants to replace assets, so let's delete all assets when this happens.

File: installer-resources/tools/uploadghasset.rb [hunk 7]
```diff
@@ -1,32 +1,22 @@
...
 
 @client = Octokit::Client.new(:access_token => ENV['GH_TOKEN'])
 
-
-def release(file, tag) 
+def release(file, tag)
   releases = @client.releases "#{USER}/#{PROJECT}"
   target_release = releases.select { |r| r.tag_name == tag }[0]
 
-  assets = @client.release_assets(target_release.url)
```
Comment (xiam) [inline installer-resources/tools/uploadghasset.rb:22]: Code moved to `createrelease.rb`, deleting assets at this point was causing only the asset added at the end to be shown.

File: releasebeta.bash [hunk 8]
```diff
@@ -1,33 +0,0 @@
-#!/usr/bin/env bash
```
Comment (xiam) [inline releasebeta.bash:1]: Integrated into Makefile: `make release-beta`

File: releaseqa.bash [hunk 9]
```diff
@@ -1,62 +0,0 @@
-#!/usr/bin/env bash
```
Comment (xiam) [inline releaseqa.bash:1]: Integrated into Makefile: `make release-qa`

File: tagandbuild.bash [hunk 10]
```diff
@@ -1,28 +0,0 @@
-#!/bin/bash
```
Comment (xiam) [inline tagandbuild.bash:1]: Integrated into Makefile. `make packages tag-and-build`

File: updateicons.bash [hunk 11]
```diff
@@ -1,3 +0,0 @@
-#!/bin/bash
```
Comment (xiam) [inline updateicons.bash:1]: Integrated into Makefile. `make update-icons`

**Comment by xiam** [comment]
Hello @myleshorton, do you think you can help me reviewing this PR? Thanks.

**Comment by myleshorton** [comment]
Of course @xiam! Checking it out now.

File: Makefile [hunk 12]
```diff
@@ -224,24 +263,19 @@ package-windows: require-version windows-386
 	docker run -v $$PWD:/flashlight-build -v $$SECRETS_DIR:/secrets -t $(DOCKER_IMAGE_TAG) /bin/bash -c 'cd /flashlight-build && BNS_CERT="/secrets/bns_cert.p12" BNS_CERT_PASS="'$$BNS_CERT_PASS'" VERSION="'$$VERSION'" make docker-package-windows' && \
 	echo "-> lantern-installer.exe"
 
-package-darwin: require-version darwin
+package-darwin: require-version require-npm require-appdmg require-svgexport darwin
```
Comment (myleshorton) [inline Makefile]: Do we really require npm here?

Comment (myleshorton) [inline Makefile]: Never mind -- yeah we run npm directly.

Comment (myleshorton) [inline Makefile]: Oh sorry -- was looking at the deleted code =). Yeah looks like maybe we no longer directly need npm.

File: Makefile [hunk 13]
```diff
@@ -253,10 +287,59 @@ binaries: docker genassets linux windows darwin
...
+	$(RUBY) ./installer-resources/tools/uploadghasset.rb $(GH_USER) $(GH_RELEASE_REPOSITORY) $$TAG update_linux_amd64.bz2 && \
+	$(RUBY) ./installer-resources/tools/uploadghasset.rb $(GH_USER) $(GH_RELEASE_REPOSITORY) $$TAG update_linux_386.bz2
+
+release-beta:
+	@BASE_NAME="lantern-installer-qa" && \
+	BETA_BASE_NAME="lantern-installer-beta" && \
+	for NAME in $$(ls -1 $$BASE_NAME.exe $$BASE_NAME.dmg $$BASE_NAME-32-bit.deb $$BASE_NAME-64-bit.deb); do \
+		BETA=$$(echo $$NAME | sed s/"$$BASE_NAME"/$$BETA_BASE_NAME/) && \
+		echo "Copying alpha $$NAME to beta $$BETA..." && \
+		s3cmd cp s3://$(S3_BUCKET)/$$NAME s3://$(S3_BUCKET)/$$BETA; \
```
Comment (myleshorton) [inline Makefile]: Use $(S3CMD) instead?

**Comment by xiam** [comment]
Thanks @myleshorton! Good observations. Care to check again?

**Comment by xiam** [comment]
Oh God. I merged by mistake. Sorry, I thought I was on https://github.com/getlantern/flashlight-build/pull/166.

**Comment by xiam** [comment]
@myleshorton Not sure what should we do. I guess we can revert if we see something unusual happening?

**Comment by myleshorton** [comment]
Let's leave it. I was pretty much done with the review anyway.

**Comment by xiam** [comment]
Cool, thanks for reviewing!

**Pull request merged**